x-pack/plugin/repository-encrypted/src/test/java/org/elasticsearch/repositories/encrypted/ChainingInputStreamTests.java

```diff
@@ -0,0 +1,573 @@
+                    return null;
+                }
+            }
+
+            @Override
+            public boolean markSupported() {
+                return false;
+            }
+        };
+        byte[] b = test.readAllBytes();
+        assertThat(b.length, Matchers.is(b1.length));
+        for (int i = 0; i < b.length; i++) {
+            Assert.assertThat(b[i], Matchers.is(b1[i]));
+        }
+        assertThat(chainComponents[0], Matchers.nullValue());
+        assertThat(chainComponents[1], Matchers.is(sourceComponents[0]));
+        assertThat(chainComponents[1].closed.get(), Matchers.is(true));
+        assertThat(chainComponents[2], Matchers.nullValue());
+        assertThat(chainComponents[3], Matchers.nullValue());
+    }
+
+    public void testCallsForwardToCurrentComponent() throws Exception {
+        InputStream mockCurrentIn = mock(InputStream.class);
+        ChainingInputStream test = new ChainingInputStream() {
+            @Override
+            InputStream nextComponent(InputStream currentComponentIn) throws IOException {
+                throw new IllegalStateException();
+            }
+        };
+        test.currentIn = mockCurrentIn;
+        // verify "byte-wise read" is proxied to the current component stream
+        when(mockCurrentIn.read()).thenAnswer(invocationOnMock -> Randomness.get().nextInt(256));
+        test.read();
+        verify(mockCurrentIn).read();
+        // verify "array read" is proxied to the current component stream
+        when(mockCurrentIn.read(org.mockito.Matchers.<byte[]>any(), org.mockito.Matchers.anyInt(), org.mockito.Matchers.anyInt())).
+                thenAnswer(invocationOnMock -> {
+                    final int len = (int) invocationOnMock.getArguments()[2];
+                    if (len == 0) {
+                        return 0;
+                    } else {
+                        // partial read return
+                        int bytesCount = 1 + Randomness.get().nextInt(len);
+                        return bytesCount;
+                    }
+                });
+        byte[] b = new byte[2 + Randomness.get().nextInt(32)];
+        int len = 1 + Randomness.get().nextInt(b.length - 1);
+        int offset = Randomness.get().nextInt(b.length - len);
+        test.read(b, offset, len);
+        verify(mockCurrentIn).read(Mockito.eq(b), Mockito.eq(offset), Mockito.eq(len));
+        // verify "skip" is proxied to the current component stream
+        long skipCount = 1 + Randomness.get().nextInt(3);
+        test.skip(skipCount);
+        verify(mockCurrentIn).skip(Mockito.eq(skipCount));
+        // verify "available" is proxied to the current component stream
+        test.available();
+        verify(mockCurrentIn).available();
+    }
+
+    public void testEmptyReadAsksForNext() throws Exception {
+        InputStream mockCurrentIn = mock(InputStream.class);
+        when(mockCurrentIn.markSupported()).thenAnswer(invocationOnMock -> true);
```

Comment:
Why `thenAnswer` rather than `thenReturn` ?

Reply:
No reason (brainless copy-pasting). I have made the change.